WordPress/wordpress-develop
# Block Hooks API: Add ignoredHookedBlocks meta when preparing REST API response

The block hooks toggle relies on the `ignoredHookedBlocks` when determining whether to show the toggle in the Site Editor UI. The current problem is that for uncustomized templates we don't add this meta data so to fix this problem we need to run the block hooks logic with the `set_ignored_hooked_blocks_metadata` callback when preparing the template for the response.

TODO:
- [x] Add filter to `prepare_item_for_response()` method
- [x] Add filter callback to run block hooks logic and add metadata attributes

Trac ticket: https://core.trac.wordpress.org/ticket/59574

**Testing instructions**:
1. Add the below code to your themes `functions.php` file to add a hooked block.
2. Ensure the Single Posts template has all customizations cleared
3. Load the Single Posts template in the Site Editor and click the Post Content block to confirm it has the hooked block toggle.

```php
function register_hooked_block( $hooked_blocks, $position, $anchor_block, $context ) {
	if ( $anchor_block === 'core/post-content' && $position === 'after' ) {
		$hooked_blocks[] = 'core/loginout';
	}

	return $hooked_blocks;
}

add_filter( 'hooked_block_types', 'register_hooked_block', 10, 4 );
```

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**
